pkg/security/tests/mount_test.go

```diff
@@ -65,19 +65,18 @@ func TestMount(t *testing.T) {
 		err = test.GetProbeEvent(func() error {
 			// Test mount
 			if err := syscall.Mount(mntPath, dstMntPath, "bind", syscall.MS_BIND, ""); err != nil {
-				t.Fatalf("could not create bind mount: %s", err)
+				return fmt.Errorf("could not create bind mount: %s", err)
 			}
 			return nil
 		}, func(event *sprobe.Event) bool {
-			assert.Equal(t, "mount", event.GetType(), "wrong event type")
-			assert.Equal(t, "/"+dstMntBasename, event.Mount.MountPointStr, "wrong mount point")
-			assert.Equal(t, "xfs", event.Mount.GetFSType(), "wrong mount fs type")
-
 			mntID = event.Mount.MountID
-			return true
+
+			return assert.Equal(t, "mount", event.GetType(), "wrong event type") &&
```

Comment:
Can't we use your new way of doing the callback here ?

Reply:
Nop this is `GetProbeEvent` => this handler is called for all mount events, until we either get the right one or timeout.